pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,752 @@
+        """
+        The original factor potentials gives us one of the integer solutions.
+        """
+        int_val = 0
+        for i in range(len(self.m_all_regions)):
+            tmpvec = []
+            base_sizes = []
+            for j in self.m_all_regions[i].rvariables_inds:
+                tmpvec.append(self.m_decoded_res[j])
+                base_sizes.append(self.m_var_sizes[j])
+            int_val += self.m_region_lambdas[i][self._get_flat_ind(tmpvec, base_sizes)]
+        return int_val
+
+    def _local_decode(self):
+        """
+        Finds the best assignment from a set of assignment by locally maximizing the single node beliefs
+        """
+        obj = 0
+        for i in range(len(self.objective)):
+            obj += max(self.objective[i])
+            index_of_maximum = self.objective[i].index(max(self.objective[i]))
+            if len(self.m_all_intersect[i]) == 1:
+                self.m_decoded_res[self.m_all_intersect[i][0]] = index_of_maximum
+
+        # update the m_best_val if it is smaller than int_val
+        int_val = self._integer_val()
+        print('int_val: "{}"'.format(int_val))
+        if int_val > self.m_best_val:
+            self.m_best_decoded_res = self.m_decoded_res
+            self.m_best_val = int_val
+        return obj
+
+    def run_mplp(self, niter, obj_del_thr, int_gap_thr):
+        """
+        Perform the GMPLP updates.
+        """
+        for i in range(niter):
+            # For each of the 24 regions
+            for r in self.m_all_regions:
+                self.objective = self._update_msgs(self.objective, r)
+            obj = self._local_decode()
+            obj_del = self.last_obj - obj
+            self.last_obj = obj
+            int_gap = obj - self.m_best_val
+
+            print("m_best_val: {}".format(self.m_best_val))
+            print("int_gap: {}".format(int_gap))
+            print("objective: {}".format(obj))
+
+            if (obj_del < obj_del_thr) and (i > len(self.m_var_sizes)):
+                break
+            if int_gap < int_gap_thr:
+                break
+
+    def _break_list(self, big_message, region):
+        """
+        Break the region's message into sub-messages for each of it's intersection
+        """
+        lst = [list(i) for i in itertools.product(*[list(range(k)) for k in region.region_var_sizes])]
+        subsets = [[-1e09] * len(i) for i in region.m_msgs_from_region]
+        no_subsets = len(region.rintersects_relative_pos)
+        for vi in range(len(big_message)):
+            for si in range(no_subsets):
+                ind = self._get_idx(lst[vi], region.rintersects_relative_pos[si], region)
+                subsets[si][ind] = max(big_message[vi], subsets[si][ind])
+        return subsets
+
+    def _get_idx(self, bit_value_big, active_index, region):
+        """
+        Converts the bit value's active indexes into decimal number corresponding to the region's cardinality.
+        For eg:
```

Comment:
Please stick to the documentation format used everywhere. Do you even look at the code written elsewhere?